File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cs
using System;

public class Solution 
{
    public int solution(string myString, string pat) 
    {
        string str = "";
        int count = 0;
        int Length = myString.Length-pat.Length;
        
        for(int i=0; i<=Length; i++) {
            str = myString.Substring(i, pat.Length);
            if (str.Contains(pat)) {
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) 
{
    size_t pos = myString.find(pat);
    int count = 0;
    while (pos != string::npos) 
    {
        count++;
        pos = myString.find(pat, pos + 1); // 다음 위치부터 다시 검색
    }
    return count;
}